### PR TITLE
Add label to Product items in menu

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
Support for adding label to Product menu items

Update Header storybook to better match current design

Screenshot 2023-03-23 at 15 44 17

Screenshot 2023-03-23 at 15 34 27

## Justify why they are needed
Highlight products 
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2030** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2030" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2030?utm_source=stack-comment).

File: packages/ui/src/components/HeadingLabel/HeadingLabel.tsx [hunk 1]
```diff
@@ -4,10 +4,10 @@ import { getMargins, Margins } from '../../lib/margins'
 import { UIColors } from '../../lib/theme/colors/colors'
 import { getColor, theme } from '../../lib/theme/theme'
 
-type HeadingLabelColors = Pick<UIColors, 'blueFill1' | 'blueFill2'>
+type HeadingLabelColors = Pick<UIColors, 'blueFill1' | 'blueFill2' | 'green50'>
 
 export type HeadingLabelProps = Margins & {
-  as?: 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+  as?: 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span'
```
Comment (guilhermespopolin) [inline packages/ui/src/components/HeadingLabel/HeadingLabel.tsx:10]: I see the reasoning behind leverage existing styles but it feels odd get that label/tag done using a component called `Heading<something>` no?! I mean, maybe it's not so important. Maybe it's just OCD.

**Review by guilhermespopolin — APPROVED** [review]

**Comment by gustaveen** [comment]
@gustaveen queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2030).

**Comment by gustaveen** [comment]
@gustaveen merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2030).

**Pull request merged**